UnityProject/Assets/Scripts/Weapons/FireLemon.cs

```diff
@@ -0,0 +1,221 @@
+	public Explosion explosionPrefab5;
+
+	[TooltipAttribute("If the fuse is precise or has a degree of error equal to fuselength / 4")]
+	public bool unstableFuse = false;
+	[TooltipAttribute("fuse timer in seconds")]
+	public float fuseLength = 3;
+
+	[Tooltip("SpriteHandler used for blinking animation")]
+	public SpriteHandler spriteHandler;
+
+	// Zero and one sprites reserved for left and right hands
+	private const int LOCKED_SPRITE = 2;
+	private const int ARMED_SPRITE = 3;
+
+	private GrownFood GrownFood;
+
+	private int lemonPotency;
+
+	///Getting Grownfood so we can get the potency of the plant.
+	private void Awake()
+	{
+		GrownFood = GetComponent<GrownFood>();
+		lemonPotency = GrownFood.plantpotency;
+	}
+
+	//whether this object has exploded
+	private bool hasExploded;
+
+	// is timer finished or was interupted?
+	[SyncVar(hook = nameof(UpdateTimer))]
+	private bool timerRunning = false;
+
+	//this object's registerObject
+	private RegisterItem registerItem;
+	private ObjectBehaviour objectBehaviour;
+
+	private void Start()
+	{
+		registerItem = GetComponent<RegisterItem>();
+		objectBehaviour = GetComponent<ObjectBehaviour>();
+
+		// Set grenade to locked state by default
+		UpdateSprite(LOCKED_SPRITE);
+	}
+
+	public void OnDespawnServer(DespawnInfo info)
+	{
+		// Set grenade to locked state by default
+		UpdateSprite(LOCKED_SPRITE);
+		// Reset grenade timer
+		timerRunning = false;
+		hasExploded = false;
+	}
+
+	public void ClientPredictInteraction(HandActivate interaction)
+	{
+		// Toggle the throw action after activation
+		UIManager.Action.Throw();
+	}
+
+	public void ServerRollbackClient(HandActivate interaction)
+	{
+	}
+
+	public void ServerPerformInteraction(HandActivate interaction)
+	{
+		if (timerRunning)
+			return;
+
+		// Toggle the throw action after activation
+		if (interaction.Performer == PlayerManager.LocalPlayer)
+		{
+			UIManager.Action.Throw();
+		}
+
+		// Start timer
+		StartCoroutine(TimeExplode(interaction.Performer));
+	}
+
+	private IEnumerator TimeExplode(GameObject originator)
+	{
+		if (!timerRunning)
+		{
+			timerRunning = true;
+			PlayPinSFX(originator.transform.position);
+
+			if (unstableFuse)
+			{
+				float fuseVariation = fuseLength / 4;
+				fuseLength = Random.Range(fuseLength - fuseVariation, fuseLength + fuseVariation);
+			}
+
+			yield return WaitFor.Seconds(fuseLength);
+
+			// Is timer still running?
+			if (timerRunning)
+				Explode();
+		}
+	}
+
+	private void UpdateSprite(int sprite)
+	{
+		// Update sprite in game
+		spriteHandler?.ChangeSprite(sprite);
+	}
+
+	public void Explode()
+	{
+		if (hasExploded)
+		{
+			return;
+		}
+		hasExploded = true;
+
+		if (isServer)
+		{
+			// Get data from grenade before despawning
+			var explosionMatrix = registerItem.Matrix;
+			var worldPos = objectBehaviour.AssumedWorldPositionServer();
+
+			// Despawn grenade
+			Despawn.ServerSingle(gameObject);
+
+			// Explosion here
+			if(lemonPotency < 30)
```

Comment:
Again, this if is very unrobust and would make some pain if someone were to add/remove another explosion prefab, or decided to change lemonPotency.
I'd suggest to use arrays and calculate index with something like
`explosionIndex = explosionPrefabs.count * (plantPotency / potencyForMaxExplosion)`